lib/workspaces/update-workspaces.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const Arborist = require('@npmcli/arborist')
+const reifyFinish = require('../utils/reify-finish.js')
+
+async function updateWorkspaces ({
+  config,
```

Comment:
it looks like we only ever pass in a dereferenced config, flatOptions, and localPrefix from npm itself, but we're already passing in npm.

Reply:
my idea here was to uncouple a bit more from the catch all `npm` object while also better highlighting what are the external objects the function is depending on - I'm not really attached to that choice and I can also change it to a more simplistic `updateWorkspaces(npm)` if you prefer